types/json/binary_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"fmt"
+
+	. "github.com/pingcap/check"
+)
+
+func (s *testJSONSuite) TestBinaryJSONMarshalUnmarshal(c *C) {
```

Comment:
Add some cases to test data marshaled by old `JSON` can be unmarshaled by `BinaryJSON` success?